### PR TITLE
Mode

### PR DESCRIPTION
**Pull request by Lancelot03**
Added mode fun in torch and fixed some formatting errors

### CODE REVIEW

**Comment by vayuusaee** [comment]
So, when you run flake8 $(git ls-files '*.py'), Git lists all Python files in the repository and passes them as arguments to Flake8, which then lints the code and reports any issues or errors.

Using this command can help ensure that your Python code adheres to the PEP 8 style guide and can catch potential issues early in the development process.

**Comment by Lancelot03** [comment]
Hi, @vayuusaee there are many lint fault in test functions.

File: ivy/functional/frontends/torch/reduction_ops.py [hunk 1]
```diff
@@ -182,3 +182,8 @@ def quantile(input, q, dim=None, keepdim=False, *, interpolation="linear", out=N
 @to_ivy_arrays_and_back
 def count_nonzero(input, dim=None):
     return ivy.count_nonzero(input, axis=dim).astype(ivy.int64)
+
+
+@to_ivy_arrays_and_back
+def mode(input, dim, keepdim=False, *, out=None):
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/reduction_ops.py:188]: should be `dim=-1` by default as in the original torch function

File: ivy/functional/frontends/torch/reduction_ops.py [hunk 2]
```diff
@@ -182,3 +182,8 @@ def quantile(input, q, dim=None, keepdim=False, *, interpolation="linear", out=N
 @to_ivy_arrays_and_back
 def count_nonzero(input, dim=None):
     return ivy.count_nonzero(input, axis=dim).astype(ivy.int64)
+
+
+@to_ivy_arrays_and_back
+def mode(input, dim, keepdim=False, *, out=None):
+    return ivy.mode(input, axis=dim, keepdims=keepdim, out=out)
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/reduction_ops.py:189]: I don't think we've got a function named `mode` in our api. This is causing the tests to fail due to unresolved reference.

Comment (Lancelot03) [reply ivy/functional/frontends/torch/reduction_ops.py]: Then how to implement it.

Comment (Lancelot03) [reply ivy/functional/frontends/torch/reduction_ops.py]: Or attached it with Tensor.mode function

Comment (AnnaTz) [reply ivy/functional/frontends/torch/reduction_ops.py]: Mode simply finds the most common value in each row of the input tensor. This is something you can implement using a composition of existing ivy functions.

Comment (Lancelot03) [reply ivy/functional/frontends/torch/reduction_ops.py]: Ok understand thanks

**Comment by Lancelot03** [comment]
#13241

**Comment by Lancelot03** [comment]
I'm going to discard all the commit and open new pr

**Pull request closed without merging**